needy/cache.py

```diff
@@ -0,0 +1,29 @@
+
+class Cache:
+    def __init__(self, configuration={}):
+        self.__configuration = configuration
+
+    @staticmethod
+    def type():
+        '''type of cache'''
+        raise NotImplementedError('type')
+
+    def description(self):
+        '''uri, path, or other configuration description'''
+        raise NotImplementedError('description')
+
+    def configuration(self):
+        '''configuration dictionary to serialize to configuration files'''
+        return self.__configuration
+
+    def archive(self, source, key):
+        '''make files and folders at source retrievable with the given key'''
+        raise NotImplementedError('archive')
+
+    def unarchive(self, destination, key):
```

Comment:
So it sounds like you're suggesting four methods in the interface: load/store file/directory.

It appears that the implementation for this would be the same for both methods, but it would make more sense, semantically, if they were separated for other implementations. Is that about right?